### PR TITLE
revised window calcs and label combining; issues #77 #78 #80

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
@subutai please review
- issue #77 : window amount (for each data file) increased to 10% from 7%
- issue #78 : overlapping windows are now merged into one larger window, taking the first and last timestamps of the union
- issue #80 : the raw labels that don't qualify for the combined labels set (i.e. their bucket size does not exceed the threshold) are printed to the console (along with the respective filename)
- additional: 
    - added `verbosity` variable for the user to set the amount of labeling metrics printed out
    - combined_labels.json is the new set of ground truth labels
    - `addKnownLabels` method at the end of labeler.py is for manually adding labels

### CODE REVIEW

File: nab/labeler.py [hunk 1]
```diff
@@ -266,14 +276,22 @@ def combineRawLabels(self):
 
       # Merge the bucketed timestamps that qualify as true anomalies
       for bucket in rawAnomalies:
+#        import pdb; pdb.set_trace()
```
Comment (subutai) [inline nab/labeler.py]: Should get rid of this in committed code.

File: nab/labeler.py [hunk 2]
```diff
@@ -312,3 +336,42 @@ def relaxWindows(self):
...
+#      print self.combinedRelaxedWindows
+      num_windows = len(windows)
+      if num_windows > 1:
+        for i in range(num_windows-1):
+          if (pandas.to_datetime(windows[i+1][0])
+              - pandas.to_datetime(windows[i][1])).total_seconds() <= 0:
+            windows[i] = [windows[i][0], windows[i+1][1]]
+            del windows[i+1]
+#            import pdb; pdb.set_trace()
+#      print "------2-------"
```
Comment (subutai) [inline nab/labeler.py]: Should these print statements be conditioned on self.verbosity?

File: nab/labeler.py [hunk 3]
```diff
@@ -312,3 +336,42 @@ def relaxWindows(self):
...
+#      print relativePath
+#      print self.combinedRelaxedWindows
+#      self.combinedRelaxedWindows[relativePath] = windows
+#      print "------3-------"
+#      print relativePath
+#      print self.combinedRelaxedWindows
+#    import pdb; pdb.set_trace()
```
Comment (subutai) [inline nab/labeler.py]: Get rid of import pdb statements in committed code.

**Comment by subutai** [comment]
Left a few comments. :+1: otherwise.

**Comment by BoltzmannBrain** [comment]
Sorry, I really thought I had removed those! Thanks @subutai

**Pull request merged**